### PR TITLE
Normalize the HTTP RESTful API

### PR DESCRIPTION
**Pull request by JiamingMai**
Normalize the HTTP RESTful API:
1) the list files API: `curl -X GET "http://<worker_ip>:<port>/v1/files?path=<path>"`
image

Now the server will also return the lengths of the files.

2) the get page API: `curl GET "http://<worker_ip>:<port>/v1/file/<file_id>/page/<page_index>"`
image

### CODE REVIEW

File: dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  /**
+   * Parse the HTTP request URI and extract the fields.
+   *
+   * @param requestUri the HTTP request URI
+   * @return a list of fields
+   */
+  public static List<String> extractFieldsFromHttpRequestUri(String requestUri) {
+    int paramIndex = requestUri.indexOf("?");
```
Comment (ChunxuTang) [inline dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java:49]: Just a quick question: Could the fileId contain a question mark(?)?

**Review by ChunxuTang — APPROVED** [review]
LGTM. Thanks for the work!

FYI: @LuQQiu

File: dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java (hunk 1 above)
Comment (JiamingMai) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java:49]: File ID cannot contain a question mark(?). This is just for parsing a URI with parameters. For example, we may need to parse this URI `http://127.0.0.1:28080/v1/file/5f2829f08879b0e89d07174cffa8d891bdf08ba9e91218e30fe39503dd42e32c/page/0?offset=5&length=7` in the future.

Comment (beinan) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java:49]: I think file id should be a md5 or sha256 hashed value, am I right? @JiamingMai so there should be no question mark.

Comment (beinan) [reply dora/core/server/worker/src/main/java/alluxio/worker/http/HttpRequestUtil.java:49]: But the dir path might contains a question mark, I think we should escape or encode the question mark in the path

**Review by beinan — APPROVED** [review]

**Review by beinan — APPROVED** [review]
lgtm, I think for the corner cases such as the question mark in the file path, we can handle them later in another PR when there is any real requirement

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Comment by JiamingMai** [comment]
The HTTP GET request requires that the string after the question mark in the URL represents the parameters. If there is a question mark elsewhere, it should be converted into an escape character.

**Comment by JiamingMai** [comment]
I found the limitations stated in the Alluxio docs (https://docs.alluxio.io/ee/user/stable/en/administration/Troubleshooting.html).

image

**Pull request merged**